versioned_docs/version-2.0/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection.mdx

```diff
@@ -0,0 +1,42 @@
+---
+sidebar_position: 5
+---
+
+import {
+  CodeTabs,
+  PythonBlock,
+  TypeScriptBlock,
+} from "@site/src/components/InstructionsWithCode";
+
+# Dynamic few shot example selection
+
+:::note
+This feature is currently in closed beta. Please sign up [here](https://forms.gle/in9R6t9HNSYMBt7P7) for access
+:::
+
+Configure your datasets so that you can search for few shot examples based on an incoming request.
+
+## Pre-conditions
+
+1. Your dataset must have the KV store data type (we do not currently support chat model or LLM type datasets)
+2. You must have an input schema defined for your dataset. See our docs on setting up schema validation [in our UI](./manage_datasets_in_application#dataset-schema-validation) for details.
+3. You must be enabled for the closed beta
+4. You must be on LangSmith cloud
+
+## Index your dataset to be searched
+
+On the datasets UI, click the `Few-shot Index` on the top right corner and hit `Start Sync`.
+
+![](../static/few-shot-index.png)
+
+This process will start to index your data to be searchable in the background. A note will appear on the
+modal above that says if your index is up to date, and if not, what version of your dataset it last indexed.
```

Comment:
So would this modal get updated once the background process is done, or need to refresh the page?

Reply:
It polls every 10 seconds:

```
 const { data: syncStatus, error } = useGetSyncStatus(datasetId ?? '', {
    refreshInterval: 10000,
    skip: !fewShotEnabled,
  });
```